packages/api/src/router/environment-policy.ts

```diff
@@ -0,0 +1,479 @@
+
+import {
+  and,
+  buildConflictUpdateColumns,
+  eq,
+  inArray,
+  isNull,
+  sql,
+  takeFirst,
+  takeFirstOrNull,
+} from "@ctrlplane/db";
+import {
+  createEnvironmentPolicy,
+  createEnvironmentPolicyDeployment,
+  environment,
+  environmentPolicy,
+  environmentPolicyApproval,
+  environmentPolicyDeployment,
+  environmentPolicyReleaseChannel,
+  environmentPolicyReleaseWindow,
+  job,
+  release,
+  releaseChannel,
+  releaseJobTrigger,
+  setPolicyReleaseWindow,
+  updateEnvironmentPolicy,
+  user,
+} from "@ctrlplane/db/schema";
+import {
+  cancelOldReleaseJobTriggersOnJobDispatch,
+  dispatchReleaseJobTriggers,
+  isPassingAllPolicies,
+} from "@ctrlplane/job-dispatch";
+import { Permission } from "@ctrlplane/validators/auth";
+import { JobStatus } from "@ctrlplane/validators/jobs";
+
+import { createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const policyRouter = createTRPCRouter({
+  deployment: createTRPCRouter({
+    bySystemId: protectedProcedure
+      .meta({
+        authorizationCheck: ({ canUser, input }) =>
+          canUser
+            .perform(Permission.SystemGet)
+            .on({ type: "system", id: input }),
+      })
+      .input(z.string().uuid())
+      .query(({ ctx, input }) =>
+        ctx.db
+          .select()
+          .from(environmentPolicyDeployment)
+          .innerJoin(
+            environmentPolicy,
+            eq(environmentPolicy.id, environmentPolicyDeployment.policyId),
+          )
+          .where(eq(environmentPolicy.systemId, input))
+          .then((d) => d.map((d) => d.environment_policy_deployment)),
+      ),
+
+    create: protectedProcedure
+      .meta({
+        authorizationCheck: ({ canUser, input }) =>
+          canUser
+            .perform(Permission.SystemUpdate)
+            .on({ type: "environment", id: input.environmentId }),
+      })
+      .input(createEnvironmentPolicyDeployment)
+      .mutation(({ ctx, input }) =>
+        ctx.db
+          .insert(environmentPolicyDeployment)
+          .values([input])
+          .returning()
+          .then(takeFirst),
+      ),
+
+    delete: protectedProcedure
+      .meta({
+        authorizationCheck: ({ canUser, input }) =>
+          canUser
+            .perform(Permission.SystemUpdate)
+            .on({ type: "environment", id: input.environmentId }),
+      })
+      .input(
+        z.object({
+          policyId: z.string().uuid(),
+          environmentId: z.string().uuid(),
+        }),
+      )
+      .mutation(({ ctx, input }) =>
+        ctx.db
+          .delete(environmentPolicyDeployment)
+          .where(
+            and(
+              eq(environmentPolicyDeployment.policyId, input.policyId),
+              eq(
+                environmentPolicyDeployment.environmentId,
+                input.environmentId,
+              ),
+            ),
+          )
+          .returning()
+          .then(takeFirst),
+      ),
+  }),
+
+  approval: createTRPCRouter({
```

Comment:
I though we had all these endpoints

Reply:
I moved the environment policy stuff into its own file because it was getting too big